ra/registration-authority.go

```diff
@@ -116,7 +116,7 @@ func (ra *RegistrationAuthorityImpl) NewRegistration(init core.Registration) (re
 	// Store the authorization object, then return it
 	reg, err = ra.SA.NewRegistration(reg)
 	if err != nil {
-		err = core.InternalServerError(err.Error())
+		err = core.MalformedRequestError(err.Error())
```

Comment:
Agreed. By using MalformedRequestError we are stating to the WFE that the error is safe to pass to the user. So we should probably replace this with a specific string (and log the error). There are a couple of other instances below.